src/cider/nrepl/middleware/info.clj

```diff
@@ -68,10 +105,9 @@
                              {:env env
                               :dialect :cljs}))]
     (cond
+      java? (info/info-java class (or member sym))
       (and ns sym) (info/info* info-params)
-      (and class member) (info/info-java class member)
-      :else (throw (Exception.
```

Comment:
The `throw` wasn't compatible with the new logic. nil is more flexible and will result in a graceful op-specific reply (e.g. status `no-info` or `no-eldoc`)